packages/core/src/stream/sse-stream.ts

```diff
@@ -44,16 +44,11 @@ export class SseStream<Item> implements AsyncIterable<Item> {
           try {
             data = JSON.parse(sse.data);
           } catch (e: any) {
-            logger.error(`Could not parse message into JSON: ${sse.data}`);
-            logger.error(`From chunk: ${sse.raw}`);
-            throw e;
+            throw new ErrorWithCause('Could not parse message into JSON', e);
           }
 
           if (data?.error) {
-            logger.error(
-              `Error received from server: ${JSON.stringify(data.error)}`
-            );
-            throw new Error(JSON.stringify(data.error));
+            throw new Error(`Error received from the server.\n${JSON.stringify(data.error)}`);
```

Comment:
I can only throw error here, because there is no error thrown by anything else, different from the case when AxiosError is thrown by Axios, we can directly pass the Axios error to ErrorWithCause.